Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { AdalService } from 'adal-angular4';
+import TsiClient from 'tsiclient';
+import { environment } from 'src/environments/environment';
+import { FacilityService } from '../services/facility.service';
+
+@Component({
+  selector: 'app-tsi-chart',
+  templateUrl: './tsi-chart.component.html',
+  styleUrls: ['./tsi-chart.component.css']
+})
+export class TsiChartComponent implements OnInit, OnChanges {
+
+  @Input() public motionSensorIds: string[];
+  @Input() public lightSensorIds: string[];
+  @Input() public tempSensorIds: string[];
+
+  private tokenRetrieved = false;
+  private token: string;
+  private client: any;
+
+  constructor(private adalService: AdalService, private facilityService: FacilityService) {
+  }
+
+  private initializeChart() {
+    const lineChart = this.client.ux.LineChart(document.getElementById('tsichart'));
+
+    const dateTimeNowUTC = new Date();
+    const thirtyDaysBack = new Date();
+    thirtyDaysBack.setDate(dateTimeNowUTC.getDate() - Number(environment.tsiHowManyDays));
+
+    const startDate = thirtyDaysBack.toISOString();
+    const endDate = dateTimeNowUTC.toISOString();
+
+    const motionPredicate = this.buildPredicateString(this.motionSensorIds);
+    const lightPredicate = this.buildPredicateString(this.lightSensorIds);
+    const tempPredicate = this.buildPredicateString(this.tempSensorIds);
+
+    const aggregateExpressions = [];
+
+    aggregateExpressions.push(
+      new this.client.ux.AggregateExpression(
+        { predicateString: motionPredicate }, // predicate
+        { property: 'Occupied', type: 'Double' }, // measure column
+        ['avg'], // measure type,
+        { from: startDate, to: endDate, bucketSize: '30m' },  // time range
+        null, // split by value, for you probably just null
```

Comment:
We can get rid of `, for you probably just null`

Reply:
Fixed.